src/Raven.Client/Documents/Operations/Attachments/GetAttachmentsOperation.cs

```diff
@@ -0,0 +1,150 @@
+
+        private class GetAttachmentsCommand : RavenCommand<Dictionary<string, AttachmentResult>>
+        {
+            private readonly JsonOperationContext _context;
+            private readonly string _documentId;
+            private readonly IEnumerable<string> _names;
+            private readonly AttachmentType _type;
+            private readonly bool _readAll;
+
+            public GetAttachmentsCommand(JsonOperationContext context, string documentId, IEnumerable<string> names, AttachmentType type, bool readAll)
+            {
+                if (string.IsNullOrWhiteSpace(documentId))
+                    throw new ArgumentNullException(nameof(documentId));
+
+                _context = context;
+                _documentId = documentId;
+                _readAll = readAll;
+                _type = type;
+
+                if (_readAll == false)
+                    _names = names ?? throw new ArgumentNullException(nameof(names));
+
+                ResponseType = RavenCommandResponseType.Empty;
+            }
+
+            public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
+            {
+                url = $"{node.Url}/databases/{node.Database}/attachments/list?id={Uri.EscapeDataString(_documentId)}";
+
+                var request = new HttpRequestMessage
+                {
+                    Method = HttpMethods.Post,
+                    Content = new BlittableJsonContent(stream =>
+                    {
+                        using (var writer = new BlittableJsonTextWriter(_context, stream))
+                        {
+                            writer.WriteStartObject();
+
+                            writer.WritePropertyName("Type");
+                            writer.WriteString(_type.ToString());
+                            writer.WriteComma();
+                            writer.WritePropertyName("ReadAll");
+                            writer.WriteBool(_readAll);
+
+                            if (_readAll == false)
+                            {
+                                writer.WriteComma();
+                                writer.WriteArray("Names", _names);
+                            }
+
+                            writer.WriteEndObject();
+                        }
+                    })
+                };
+
+                return request;
+            }
+
+            public override async Task<ResponseDisposeHandling> ProcessResponse(JsonOperationContext context, HttpCache cache, HttpResponseMessage response, string url)
+            {
+                context.Reset();
+                context.Renew();
+
+                AttachmentsDetails attachmentsMetadata;
+                AttachmentsStreamInfo streamInfo;
+                var state = new JsonParserState();
+                Stream stream = response.Content.ReadAsStreamAsync().Result;
+                context.GetManagedBuffer(out JsonOperationContext.ManagedPinnedBuffer buffer);
```

Comment:
shouldn't this be in a using?

Reply:
I pass this buffer as part of the Result